apps/balancer-tools/src/app/apr/(utils)/tokenApr.ts

```diff
@@ -285,3 +321,159 @@ const getImplementationAddress = async (
     return undefined;
   }
 };
+
+async function calculateTokenApr({
+  tokenYield,
+  poolId,
+  chain,
+  tokenAddress,
+  tokenWeight,
+  timeStart,
+  timeEnd,
+}: {
+  tokenYield: number;
+  poolId: string;
+  chain: string;
+  tokenAddress: Address;
+  tokenWeight: number;
+  timeStart: number;
+  timeEnd: number;
+}) {
+  const rangeData = await poolsWithCache.gql(chain).poolSnapshotInRange({
+    poolId,
+    timestamp: generateDateRange(timeStart, timeEnd),
+  });
+
+  let pool: Pool;
+
+  if (rangeData.poolSnapshots.length > 0) {
+    pool = rangeData.poolSnapshots[0].pool;
+  } else {
+    const currentPoolData = await pools.gql(chain).Pool({ poolId });
```

Comment:
Just to keep in mind -- I think we would've already fetched this in that Promise.all where we get data from all APR functions